internal/ir/xds.go

```diff
@@ -2513,6 +2521,8 @@ type LoadBalancer struct {
 	Random *Random `json:"random,omitempty" yaml:"random,omitempty"`
 	// ConsistentHash load balancer policy
 	ConsistentHash *ConsistentHash `json:"consistentHash,omitempty" yaml:"consistentHash,omitempty"`
+	// ZoneAware defines the configuration related to the distribution of requests between locality zones.
+	ZoneAware *ZoneAware `json:"zoneAware,omitempty" yaml:"zoneAware,omitempty"`
```

Comment:
is this field needed in this iteration ?
trying to understand why are we not consolidating the two fields - `PreferLocal` and `ZoneAware.PreferLocal`

Reply:
Could you elaborate? Consolidating these two isn't really an option.
```
zoneAware: # maps to Envoy LocalityLbConfig
  preferLocal: {} # maps to Envoy ZoneAwareLbConfig
  # not implemented yet
  weighted: {} # maps to localityWeightedLbConfig
```

Reply:
Context: https://github.com/envoyproxy/gateway/pull/6154#discussion_r2127236959

Reply:
Specifically in the internal ir logic? Nope I can remove that if preferred but it makes the future implementation of `weighted` easier to approach.